pkg/api/plugins.go

```diff
@@ -33,12 +33,26 @@ func (hs *HTTPServer) GetPluginList(c *models.ReqContext) response.Response {
 	typeFilter := c.Query("type")
 	enabledFilter := c.Query("enabled")
 	embeddedFilter := c.Query("embedded")
+	// "" => no filter
+	// "0" => filter out core plugins
+	// "1" => filter out non-core plugins
 	coreFilter := c.Query("core")
 
+	hasAccess := ac.HasAccess(hs.AccessControl, c)
+
 	// When using access control anyone that can create a data source should be able to list all data sources installed
-	// Fallback to only letting admins list non-core plugins
-	hasAccess := accesscontrol.HasAccess(hs.AccessControl, c)
-	if !hasAccess(accesscontrol.ReqOrgAdmin, accesscontrol.EvalPermission(datasources.ActionCreate)) && !c.HasRole(models.ROLE_ADMIN) {
+	// Organization Admins & Server Admins should be able to list plugins
```

Comment:
Would you mind adding a test for these cases? Shouldn't be too tricky to add here https://github.com/grafana/grafana/blob/main/pkg/tests/api/plugins/api_plugins_test.go#L69

Reply:
Hey I will take over this due to @gamab being out of office. I looked at the test and the only way to tests this correctly would to load a non core plugin on startup. Without a non core pluing the result would be the same, [early return](https://github.com/grafana/grafana/blob/gamab/fix/plugins-list/pkg/api/plugins.go#L50-L52) when user is not org or grafana admin and if user is either grafana admin or org admin all plugins will be filtered out

Reply:
You can still write a test for that. I can help 😄 Is it okay for me to commit to this branch or shall I send a patch to you directly? I'm actually not able to repro the issue so regardless it would be good for me to understand what needs to be corrected to see the issue

Reply:
Yes it is okey if you want to do it :)

Reply:
In order to reproduce it a user needs to be Grafana admin (not org admin) and you need to remove the RBAC permission `datasource:create`. Only a issue in enterprise

Reply:
Okay have pushed a commit now. Please check it out and let me know what I'm missing as I'm unable to get that to fail on `main`

Reply:
Thanks will take a look first thing tomorrow. My guess is that Grafana Admins has the RBAC data source create permissions by default. This pr is trying to cover the case when that permission has been removed. A Grafana admin should still be able to list the plugins